library/std/src/sys_common/wtf8.rs

```diff
@@ -575,6 +627,11 @@ impl Wtf8 {
         }
     }
 
+    /// Creates an owned `Wtf8Buf` from a borrowed `Wtf8`.
+    pub fn to_owned(&self) -> Wtf8Buf {
+        Wtf8Buf { bytes: self.bytes.to_vec(), is_known_utf8: false }
```

Comment:
```suggestion
        Wtf8Buf { bytes: self.bytes.to_vec(), is_known_utf8: self.is_known_utf8 }
```

Reply:
I only added `is_known_utf8` to `Wtf8Buf`, and not `Wtf`, because `Wtf` does `transmute` tricks with `&[u8]`.

If it were important to set `is_known_utf8` here, we could do a scan here. But since the main purpose of `is_known_utf8` is to avoid doing a scan later, doing a scan eagerly in order to set it isn't a clear win.

Reply:
I'm confused. Why is scanning here necessary? There's no transformation happening here, it should always be the same as whatever `is_known_utf8` is currently set to.

Reply:
This is in `Wtf8::to_owned`, and `Wtf8` doesn't have a `is_known_utf8` field itself, and it doesn't always point at a `Wtf8Buf`. It may be pointing at an arbitrary `&[u8]` where there is no `is_known_utf8` field.